src/server/utils/prepare.js

```diff
@@ -10,21 +10,43 @@ var exec = require('child_process').exec,
  * @return {Promise}
  */
 function execCordovaPrepare(projectRoot, platform) {
-    var deferred = Q.defer();
+    return retryAsync(getExecCordovaPrepareImpl(projectRoot, platform), 2);
+}
+
+function getExecCordovaPrepareImpl(projectRoot, platform) {
+    return function () {
+        var deferred = Q.defer();
 
-    log.log('Preparing platform \'' + platform + '\'.');
+        log.log('Preparing platform \'' + platform + '\'.');
 
-    exec('cordova prepare ' + platform, {
-        cwd: projectRoot
-    }, function (err, stdout, stderr) {
-        if (err) {
-            deferred.reject(err || stderr);
+        exec('cordova prepare ' + platform, {
+            cwd: projectRoot
+        }, function (err, stdout, stderr) {
+            if (err || stderr) {
+                deferred.reject(err || stderr);
+            } else {
+                deferred.resolve();
+            }
+        });
+
+        return deferred.promise;
+    };
+}
+
+function retryAsync(promiseFunc, maxTries, delay, iteration) {
```

Comment:
We can improve the documentation here, at least of the parameters, to know which are required and which are optional.